monai/transforms/intensity/array.py

```diff
@@ -2053,3 +2055,123 @@ def __call__(self, img: NdarrayOrTensor, mask: Optional[NdarrayOrTensor] = None)
         out, *_ = convert_to_dst_type(src=ret, dst=img, dtype=self.dtype or img.dtype)
 
         return out
+
+
+class IntensityRemap(RandomizableTransform):
+    """
+    Transform for intensity remapping of images. The intensity at each
+    pixel is replaced by a new values coming from an intensity remappping
+    curve.
+
+    The remapping curve is created by uniformly sampling values from the
+    possible intensities for the input image and then adding a linear
+    component. The curve is the rescaled to the input image intensity range.
+
+    Intended to be used as a means to data augmentation via:
+    :py:class:`monai.transforms.RandIntensityRemap`.
+
+    Implementation is described in the work:
+    `Intensity augmentation for domain transfer of whole breast segmentation
+    in MRI <https://ieeexplore.ieee.org/abstract/document/9166708>`_.
+
+    Note: for large images the transform may be too slow to apply on the fly.
+
+    Args:
+        kernel_size: window size for averaging operation for the remapping
+            curve.
+        slope: slope of the linear component. Easiest to leave default value
+            and tune the kernel_size parameter instead.
+        return_map: set to True for the transform to return a dictionary version
+            of the lookup table used in the intensity remapping. The keys
+            correspond to the old intensities, and the values are the new
+            values.
+    """
+
+    def __init__(self, kernel_size: int = 30, slope: float = 0.7, return_map: bool = False):
+
+        super().__init__()
+
+        self.kernel_size = kernel_size
+        self.slope = slope
+        self.return_map = return_map
+
+    def __call__(self, img: torch.Tensor) -> Union[torch.Tensor, Tuple[torch.Tensor, Mapping]]:
+        """
+        Args:
+            img: image to remap.
+        """
+
+        img = img.clone()
+        # sample noise
+        vals_to_sample = torch.unique(img).tolist()
+        noise = torch.from_numpy(self.R.choice(vals_to_sample, len(vals_to_sample) - 1 + self.kernel_size))
+        # smooth
+        noise = torch.nn.AvgPool1d(self.kernel_size, stride=1)(noise.unsqueeze(0)).squeeze()
+        # add linear component
+        grid = torch.arange(len(noise)) / len(noise)
+        noise += self.slope * grid
+        # rescale
+        noise = (noise - noise.min()) / (noise.max() - noise.min()) * img.max() + img.min()
+
+        # intensity remapping function
+        map = dict(zip(vals_to_sample, noise.tolist()))
```

Comment:
looks good, perhaps implementing this with torch.bucketize https://pytorch.org/docs/stable/generated/torch.bucketize.html could be more efficient...

Reply:
Hi @wyli, I looked at the bucketize link you provided, and I don't see how it helps. The line `map = dict(zip(vals_to_sample, noise.tolist()))` is simply creating a lookup table such as { 'old intensity' : 'new intensity' }. Not clear to me how torch.bucketize helps here.

Reply:
the (untested) idea is that if `vals_to_sample` is sorted monotonically increasing, then `index_img = torch.bucketize(img, vals_to_sample)` will have the indices of every intensity, and then the mapping could be done by `new_img = noise[index_img]`? essentially `for key in map: index_img = (img==key)` is a "one-hot" encoding, `index_img = torch.bucketize(img, vals_to_sample)` is the corresponding non-onehot version. but maybe it doesn't work or it's even slower...

Reply:
Using `torch.bucketize` really made the code very efficient. I'll add dictionary versions of the transforms, plus unit tests.

Sorry, I keep seeing that my commit is touching other files...